WcaOnRails/app/webpacker/components/CompetitionForm/CompsTable.js

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable react/no-danger */
+import React, { useState } from 'react';
+import {
+  Button, Table, Message,
+} from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+import Loading from '../Requests/Loading';
+
+function NotConfirmedIcon() {
+  return (
+    <i
+      className="fas fa-exclamation-circle"
+      data-toggle="tooltip"
+      data-placement="top"
+      data-container="body"
+      title="This competition is not confirmed yet."
+    />
+  );
+}
+
+function CompsTableContent({ nearby, action }) {
+  const [showEvents, setShowEvents] = useState(false);
+
+  const headerRow = () => (
+    <Table.Row>
+      <Table.HeaderCell name="name" width={3}>
+        {I18n.t('competitions.adjacent_competitions.name')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="delegates" width={3}>
+        {I18n.t('competitions.adjacent_competitions.delegates')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="date" width={2}>
+        {I18n.t('competitions.adjacent_competitions.date')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="location" width={2}>
+        {I18n.t('competitions.adjacent_competitions.location')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="distance" width={1}>
+        {I18n.t('competitions.adjacent_competitions.distance')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="limit" width={1}>
+        {I18n.t('competitions.adjacent_competitions.limit')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="competitors" width={1}>
+        {I18n.t('competitions.adjacent_competitions.competitors')}
+      </Table.HeaderCell>
+      {showEvents ? (
+        <Table.HeaderCell name="events" width={2}>
+          {I18n.t('competitions.adjacent_competitions.events')}
+        </Table.HeaderCell>
+      ) : null}
+    </Table.Row>
+  );
+
+  const compRow = (comp) => (
+    <Table.Row key={comp.name} warning={!comp.danger} error={comp.danger}>
+      <Table.Cell name="name" width={3}>
+        <span dangerouslySetInnerHTML={{ __html: comp.nameLink }} />
```

Comment:
Why is this `dangerouslySetInnerHTML` used? Can it not be an `a` tag?

Reply:
comp.nameLink is passed as html through comeptitionform_nearby_json in WcaOnRails/app/controllers/competitions_controller.rb